# BRIEF PREDICTION DE LA GRAVITE DES ACCIDENTS DE LA ROUTE

## Import des librairies

In [4]:
import numpy as np
import pandas as pd

import plotly.express as px
import matplotlib.pyplot as plt
import seaborn as sns

## Usagers

### Import du jeu de données 

In [7]:
df_caract =pd.read_csv("caract-2023.csv", delimiter = ';', encoding = 'utf_8')

In [8]:
df_caract

Num_Acc  jour  mois    an   hrmn  lum  dep    com  agg  int  atm  \
0      202300000001     7     5  2023  06:00    1   75  75101    2    4    2   
1      202300000002     7     5  2023  05:30    5   94  94080    2    1    3   
2      202300000003     7     5  2023  20:50    1   94  94022    2    3    2   
3      202300000004     6     5  2023  23:57    5   94  94078    2    1    3   
4      202300000005     7     5  2023  00:50    5   94  94068    2    2    3   
...             ...   ...   ...   ...    ...  ...  ...    ...  ...  ...  ...   
54817  202300054818    26    10  2023  20:45    5  974  97416    2    1    1   
54818  202300054819    26    10  2023  19:10    3  974  97416    1    1    1   
54819  202300054820    26    10  2023  09:40    1  974  97411    2    1    1   
54820  202300054821    26    10  2023  17:20    1  973  97302    1    6    1   
54821  202300054822    20    10  2023  16:30    1   69  69387    2    1    6   

       col                           adr            lat           long  
0        7                 RUE DE RIVOLI    48,86638600     2,32347100  
1        6               Avenue de Paris    48,84547782     2,42868146  
2        1     Avenue du Général Leclerc    48,76240000     2,40655000  
3        5                  Rue de Paris    48,73248432     2,44687575  
4        3          56bis Avenue Raspail    48,78581000     2,49217000  
...    ...                           ...            ...            ...  
54817    6     LA FONTAINE (RUE JEAN DE)   -21,33828000    55,47771000  
54818    3          RN3 (ANCIENNE ROUTE)   -21,28865000    55,50994000  
54819    7  BAMBOU (CHEMIN DE LA RUELLE)   -20,90129000    55,40598000  
54820    3             ROUTE NATIONALE 1     4,89713000   -52,32854000  
54821    3          Boulevard Yves Farge    45,73306000     4,82540000  

[54822 rows x 15 columns]

In [9]:
df_caract.head(20)

Num_Acc  jour  mois    an   hrmn  lum dep    com  agg  int  atm  col  \
0   202300000001     7     5  2023  06:00    1  75  75101    2    4    2    7   
1   202300000002     7     5  2023  05:30    5  94  94080    2    1    3    6   
2   202300000003     7     5  2023  20:50    1  94  94022    2    3    2    1   
3   202300000004     6     5  2023  23:57    5  94  94078    2    1    3    5   
4   202300000005     7     5  2023  00:50    5  94  94068    2    2    3    3   
5   202300000006     7     5  2023  18:10    1  94  94058    2    3    2    6   
6   202300000007     7     5  2023  18:45    1  94  94076    2    1    2    4   
7   202300000008     7     5  2023  17:20    1  94  94011    2    2    3    1   
8   202300000009     7     5  2023  18:50    1  94  94022    2    3    8    3   
9   202300000010     7     5  2023  18:20    1  94  94079    2    1    2    6   
10  202300000011     7     5  2023  12:15    1  95  95527    1    9    2    3   
11  202300000012     6     5  2023  11:15    1  93  93051    2    3    8    3   
12  202300000013     7     5  2023  20:10    1  93  93030    2    1    2    2   
13  202300000014     7     5  2023  10:30    1  93  93013    2    6    1    6   
14  202300000015     7     5  2023  19:15    1  93  93010    2    1    2    2   
15  202300000016     7     5  2023  01:56    5  92  92050    2    2    3    3   
16  202300000017     7     5  2023  21:40    5  92  92007    2    9    2    3   
17  202300000018     7     5  2023  12:10    1  01  01383    1    3    3    1   
18  202300000019     8     5  2023  12:35    1  38  38378    2    1    1    6   
19  202300000020     8     5  2023  15:10    1  91  91022    1    1    1    1   

                                       adr          lat        long  
0                            RUE DE RIVOLI  48,86638600  2,32347100  
1                          Avenue de Paris  48,84547782  2,42868146  
2                Avenue du Général Leclerc  48,76240000  2,40655000  
3                             Rue de Paris  48,73248432  2,44687575  
4                     56bis Avenue Raspail  48,78581000  2,49217000  
5                      Avenue Gabriel Péri  48,83717000  2,51361000  
6               185 Boulevard Maxime Gorki  48,78693219  2,36756095  
7   Route de Stains / Rue du Moulin Bateau  48,78238000  2,49510000  
8                  Avenue de la République  48,76240184  2,40690601  
9                   Boulevard de Friedberg  48,83266000  2,54378000  
10                                      A3  48,99572240  2,51560628  
11            Rue du Docteur Vaillant / A4  48,83611000  2,56460000  
12                 Avenue Ambroise Croizat  48,95859534  2,41581470  
13                     Carrefour Lindbergh  48,94211000  2,42916000  
14                         Avenue Galliéni  48,90631671  2,48228681  
15       CLEMENCEAU GEORGES AVENUE N° PAIR  48,88677185  2,20519647  
16                VERDUN (RUE DE) N° PAIRS  48,80202300  2,31800600  
17                                   D1083  46,08672900  5,11126600  
18                   Route de condrieu, D4  45,43994100  4,76989000  
19                              Arrancourt  48,33290200  2,14722600

In [10]:
df_caract.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54822 entries, 0 to 54821
Data columns (total 15 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Num_Acc  54822 non-null  int64 
 1   jour     54822 non-null  int64 
 2   mois     54822 non-null  int64 
 3   an       54822 non-null  int64 
 4   hrmn     54822 non-null  object
 5   lum      54822 non-null  int64 
 6   dep      54822 non-null  object
 7   com      54822 non-null  object
 8   agg      54822 non-null  int64 
 9   int      54822 non-null  int64 
 10  atm      54822 non-null  int64 
 11  col      54822 non-null  int64 
 12  adr      53433 non-null  object
 13  lat      54822 non-null  object
 14  long     54822 non-null  object
dtypes: int64(9), object(6)
memory usage: 6.3+ MB


In [11]:
df_caract.describe()

Num_Acc          jour          mois       an           lum  \
count  5.482200e+04  54822.000000  54822.000000  54822.0  54822.000000   
mean   2.023000e+11     15.396629      6.744628   2023.0      1.937653   
std    1.582589e+04      8.705039      3.368853      0.0      1.485124   
min    2.023000e+11      1.000000      1.000000   2023.0     -1.000000   
25%    2.023000e+11      8.000000      4.000000   2023.0      1.000000   
50%    2.023000e+11     15.000000      7.000000   2023.0      1.000000   
75%    2.023000e+11     23.000000     10.000000   2023.0      3.000000   
max    2.023001e+11     31.000000     12.000000   2023.0      5.000000   

                agg           int           atm           col  
count  54822.000000  54822.000000  54822.000000  54822.000000  
mean       1.626135      2.085167      1.636296      4.034129  
std        0.483833      2.022823      1.692539      1.973341  
min        1.000000     -1.000000     -1.000000     -1.000000  
25%        1.000000      1.000000      1.000000      3.000000  
50%        2.000000      1.000000      1.000000      3.000000  
75%        2.000000      2.000000      1.000000      6.000000  
max        2.000000      9.000000      9.000000      7.000000

In [12]:
df_caract.isnull().sum()

Num_Acc       0
jour          0
mois          0
an            0
hrmn          0
lum           0
dep           0
com           0
agg           0
int           0
atm           0
col           0
adr        1389
lat           0
long          0
dtype: int64

Jeu de données pour les caractéristiques de 54822 lignes pour 15 colonnes.


In [14]:
df_caract.columns

Index(['Num_Acc', 'jour', 'mois', 'an', 'hrmn', 'lum', 'dep', 'com', 'agg',
       'int', 'atm', 'col', 'adr', 'lat', 'long'],
      dtype='object')

### Suppression des colonnes inutiles


In [16]:
df_caract.drop(['hrmn', 'adr', 'an', 'jour', 'lat', 'long'], axis=1, inplace=True)

### Observation des variables

In [18]:
# Mois
df_caract['mois'].value_counts(normalize = True)

mois
6     0.099449
10    0.098300
9     0.094141
11    0.088158
5     0.086954
7     0.086717
12    0.081172
4     0.075918
8     0.075171
1     0.073930
3     0.072927
2     0.067163
Name: proportion, dtype: float64

In [19]:
# Lumière
df_caract['lum'].value_counts(normalize = True)

lum
 1    0.658805
 5    0.155011
 3    0.107858
 2    0.066451
 4    0.011838
-1    0.000036
Name: proportion, dtype: float64

In [20]:
# Création de classes
df_caract['lum'] = df_caract['lum'].replace([3,4,6], 2) # Mouillée/flaques/inondées/boue
df_caract['lum'] = df_caract['lum'].replace(7, 5)    # Verglas/neige
df_caract['lum'] = df_caract['lum'].replace(8, 9)   # Corps gras => autre

In [21]:
# Suppression des -1
df_caract = df_caract[df_caract['lum']!= -1]

In [22]:
# Département
df_caract['dep'].unique()

array(['75', '94', '95', '93', '92', '01', '38', '91', '22', '972', '974',
       '06', '67', '56', '2B', '82', '11', '63', '47', '39', '29', '26',
       '60', '45', '64', '16', '07', '17', '23', '87', '30', '34', '62',
       '987', '971', '66', '83', '69', '33', '80', '13', '973', '78',
       '976', '2A', '76', '74', '71', '72', '61', '86', '70', '65', '44',
       '54', '53', '59', '85', '57', '28', '37', '52', '51', '49', '88',
       '10', '36', '35', '31', '27', '84', '21', '08', '02', '81', '77',
       '14', '50', '42', '09', '15', '73', '32', '04', '58', '978', '68',
       '988', '55', '41', '43', '25', '05', '12', '19', '79', '40', '18',
       '03', '46', '90', '24', '89', '48', '986', '977', '975'],
      dtype=object)

In [23]:
df_caract['dep'].value_counts(normalize = True)

dep
75     0.086884
93     0.048340
92     0.046753
94     0.040204
13     0.037523
         ...   
90     0.000675
978    0.000547
986    0.000182
977    0.000128
975    0.000036
Name: proportion, Length: 107, dtype: float64

In [24]:
# Suppression de la colonne car non pertinent
df_caract.drop('dep', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_17788\1600446078.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_caract.drop('dep', axis = 1, inplace = True)


In [25]:
# Commune
df_caract['com'].value_counts(normalize = True)

com
75116    0.009230
97302    0.008045
75117    0.006640
75112    0.006385
75119    0.006348
           ...   
44091    0.000018
63393    0.000018
32305    0.000018
32020    0.000018
17083    0.000018
Name: proportion, Length: 11310, dtype: float64

In [26]:
# Suppression de la colonne car non pertinent
df_caract.drop('com', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_17788\3952229099.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_caract.drop('com', axis = 1, inplace = True)


In [27]:
# Localisation
df_caract['agg'].value_counts(normalize = True)

agg
2    0.62614
1    0.37386
Name: proportion, dtype: float64

In [28]:
# Intersection
df_caract['int'].value_counts(normalize = True)

int
 1    0.635553
 2    0.123003
 3    0.115998
 9    0.045403
 6    0.044491
 4    0.019974
 7    0.007570
 5    0.005327
 8    0.002572
-1    0.000109
Name: proportion, dtype: float64

In [29]:
# Suppression de la colonne car non pertinent
df_caract.drop('int', axis = 1, inplace = True)

C:\Users\peggy\AppData\Local\Temp\ipykernel_17788\3870245278.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_caract.drop('int', axis = 1, inplace = True)


In [30]:
# Condition atmosphérique
df_caract['atm'].value_counts(normalize = True)

atm
 1    0.779770
 2    0.120941
 8    0.040733
 3    0.027453
 7    0.017402
 5    0.004360
 9    0.003794
 6    0.003375
 4    0.002134
-1    0.000036
Name: proportion, dtype: float64

In [31]:
# Création de classes
df_caract['atm'] = df_caract['atm'].replace([3,4], 2)  # pluie/neige/grêle
df_caract['atm'] = df_caract['atm'].replace(5, 7)  # pb visibilité
df_caract['atm'] = df_caract['atm'].replace(8, 1)   # normal / couvert

C:\Users\peggy\AppData\Local\Temp\ipykernel_17788\1412711046.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_caract['atm'] = df_caract['atm'].replace([3,4], 2)  # pluie/neige/grêle
C:\Users\peggy\AppData\Local\Temp\ipykernel_17788\1412711046.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_caract['atm'] = df_caract['atm'].replace(5, 7)  # pb visibilité
C:\Users\peggy\AppData\Local\Temp\ipykernel_17788\1412711046.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice fro

In [32]:
# Suppression des -1
df_caract = df_caract[df_caract['atm']!= -1]

In [33]:
# type de collision
df_caract['col'].value_counts(normalize = True)

col
 6    0.303276
 3    0.298734
 2    0.132584
 1    0.107136
 7    0.094768
 4    0.033620
 5    0.029644
-1    0.000237
Name: proportion, dtype: float64

In [34]:
# Suppression de la colonne car non pertinent
df_caract.drop('col', axis = 1, inplace = True)

In [35]:
df_caract.info()

<class 'pandas.core.frame.DataFrame'>
Index: 54818 entries, 0 to 54821
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   Num_Acc  54818 non-null  int64
 1   mois     54818 non-null  int64
 2   lum      54818 non-null  int64
 3   agg      54818 non-null  int64
 4   atm      54818 non-null  int64
dtypes: int64(5)
memory usage: 2.5 MB


In [36]:
## Exportation du dataframe nettoyé
df_caract.to_csv('caract_clean.csv', index=False)